LSDB CI pipeline failing for Windows+python3.10

**Bug report**

When preparing a PR for this project, the CI pipeline check for Windows with Python 3.10 fails, and the failure is in the test for the crossmatch algorithm, a discrepancy in the number of crossmatches (101 vs. 103). From `def test_kdtree_crossmatch_nested(algo, small_sky_catalog, small_sky_xmatch_catalog, xmatch_correct):`

```
>       assert np.sum(xmatched["small_sky_xmatch"].nest.list_lengths) == len(xmatch_correct)
E       assert np.int64(101) == 103
```

The other 3.10 builds do not fail. 

**Before submitting**
Please check the following:

- [X] I have described the situation in which the bug arose, including what code was executed, information about my environment, and any applicable data others will need to reproduce the problem.
- [X] I have included available evidence of the unexpected behavior (including error messages, screenshots, and/or plots) as well as a description of what I expected instead.
- [ ] If I have a solution in mind, I have provided an explanation and/or pseudocode and/or task list.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -24,7 +24,7 @@ dependencies = [
     "deprecated",
     "hats>=0.6.4",
     "nested-pandas>=0.4.7,<0.5.0",
-    "pyarrow>=14.0.1",
+    "pyarrow>=14.0.1,<21.0.0",
     "scipy>=1.7.2", # kdtree
     "universal-pathlib>=0.2.2",
 ]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -24,7 +24,8 @@ dependencies = [
     "deprecated",
     "hats>=0.6.4",
     "nested-pandas>=0.4.7,<0.5.0",
-    "pyarrow>=14.0.1,<21.0.0",
+    "pyarrow>=14.0.1,<21.0.0; platform_system == 'Windows'",
+    "pyarrow>=14.0.1; platform_system != 'Windows'",
     "scipy>=1.7.2", # kdtree
     "universal-pathlib>=0.2.2",
 ]

